scripts/g.extension/g.extension.py

```diff
@@ -2218,7 +2277,10 @@ def resolve_known_host_service(url, name, branch):
         else:
             actual_start = ""
         if "branch" in match["url_end"]:
-            suffix = match["url_end"].format(name=name, branch=branch)
+            suffix = match["url_end"].format(
+                name=name,
+                branch=branch if branch else "main",  # get_default_branch(url)
```

Comment:
Was your intention to call get_default_branch for GitLab? You could return "main" from get_default_branch for GitHub.

Reply:
Actually, I was not sure myself. I solved it now by fetching the default branch for gitlab and using `main` for all other hosting services as default.